indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskMaster.java

```diff
@@ -113,7 +113,7 @@ public void becomeLeader()
         log.info("By the power of Grayskull, I have the power!");
 
         try {
-          taskLockbox.syncFromStorage();
+          SyncResult syncResult = taskLockbox.syncFromStorage();
```

Comment:
This is interesting. `TaskLockbox` is synced with storage just once when we become leader, whereas the `TaskQueue` is synced periodically?

I wonder if we shouldn't just move the invocation of `taskLockbox.syncFromStorage()` to `TaskQueue.start()`. It would help `TaskQueue` ensure that `TaskLockbox` is actually synced without having to rely on something done before it was constructed.
This would also help us avoid passing the `SyncResult` around.

The only difference I see is if `taskLockbox.syncFromStorage()` throws an exception, then currently we would fail early, while becoming leader. Whereas, if we move the sync inside `TaskQueue`, it would fail on the lifecycle start, which is effectively the same.

Reply:
Thanks! I have moved both TaskLockbox's sync and the processing to TaskQueue's start